controllers/slice/namespaces.go

```diff
@@ -391,51 +391,182 @@ func (r *SliceReconciler) unbindAppNamespace(ctx context.Context, slice *kubesli
 	if err != nil && !errors.IsNotFound(err) {
 		log.Error(err, "NS unbind: Failed to remove slice netpol", "namespace", appNs)
 	}
-	//remove the deployment annotations and labels from this namespace
+	//remove the resource annotations and labels from this namespace
 	return r.deleteAnnotationsAndLabels(ctx, slice, appNs)
 }
 
 func (r *SliceReconciler) deleteAnnotationsAndLabels(ctx context.Context, slice *kubeslicev1beta1.Slice, appNs string) error {
 	log := logger.FromContext(ctx).WithValues("type", "appNamespaces")
-	deployList := appsv1.DeploymentList{}
+	podList := corev1.PodList{}
 	listOpts := []client.ListOption{
 		client.InNamespace(appNs),
 	}
+	if err := r.List(ctx, &podList, listOpts...); err != nil {
+		log.Error(err, "Namespace offboarding:cannot list pods under ns", appNs)
+	}
+
+	if len(podList.Items) != 0 {
```

Comment:
I would not want the pod annotations and labels to be removed, because removing it still not removes the sidecars.
So it is better to keep it so that when user describes the pod , the user sees the nsm annotation.
Removing it might cause confusion for the user.

Thoughts? @bharath-avesha @kmjayadeep @YachikaRalhan

Reply:
Sure @rahulsawra98 will keep the annotations then